### ISSUE TITLE
Java IO-Exception, when enabling the plugin

### ISSUE THREAD

**Issue by eingruenesbeb** [WasabiThumb/xclaim#5]
# Description:
I'm running a test server with the latest paper build for 1.19 and the newest version of this plugin. When starting the server (after the initial install of this plugin) I encounter these warnings:
![Screenshot 2022-07-15 214417](https://user-images.githubusercontent.com/63373786/179299770-dd1a3994-6899-4123-8e07-fe6bf292ff4a.png)
I tried this both on a server with other and on one with no other plugins installed and I have not done anything in the config and just ran it as is. This seems to be a problem with the loading of the language file.

# Details:
- System Info: Java 17 (OpenJDK 64-Bit Server VM 17.0.2+8) Host: Windows 10 10.0 (amd64)
- Minecraft version: Paper 1.19 Build #61
- Plugin Version: 1.7.0
- Other Plugins: None

---
This may not cause any issues directly (as far as I can tell), but I thought I'd let you know 🙃

**Comment by WasabiThumb** [comment]
I can't tell you exactly what went wrong at this time, I'll give it a closer look later. My guess is that it doesn't have write permissions in the data directory for some reason.

**Comment by WasabiThumb** [comment]
Temporary workaround for you would be to downgrade to before the language update or just deal with the warnings if they don't cause any issues

**Comment by eingruenesbeb** [comment]
Huh... it should have read and write access...
After taking a closer look at the code myself, I think it does work with no issues, but handling the file not being there with a try except clause with an exception a bit jarring.
I think this slight rework would suffice:
```
if (bundledFile.createNewFile()) {
	  try {
	      try (InputStream is = Objects.requireNonNull(getResource("lang/" + bundled + ".json"))) {
	          JsonObject model = gson.fromJson(new InputStreamReader(is), JsonObject.class);
	          for (Map.Entry<String, JsonElement> entry : model.entrySet()) {
	              curJson.add(entry.getKey(), entry.getValue());
	          }
	      }
	      try (OutputStream os = new FileOutputStream(bundledFile)) {
	          String json = gson.toJson(curJson);
	          os.write(json.getBytes(StandardCharsets.UTF_8));
	          os.flush();
	      }
	  } catch (Exception e) {
	      e.printStackTrace();
	  }
}
```

**Comment by WasabiThumb** [comment]
That is a nice solution, however this gets ran directly beforehand to make sure it doesn't exist (we want to overwrite its contents)
```java
FileUtils.forceDelete(bundledFile)
```

**Comment by eingruenesbeb** [comment]
Ok, seems reasonable...
I'll try to run the server on a different system. And report back with my findings. It might be a weird problem with Windows Defender (although there were no reports of blocked file access).

**Comment by WasabiThumb** [comment]
Yep, I write, build and test this plugin on Linux. I bet that pebblehost also uses linux servers, so it could very well be a platform issue. Though forceDelete is an apache commons IO method so it should have great compatibility

**Comment by WasabiThumb** [comment]
I created a [new branch](https://github.com/WasabiThumb/xclaim/tree/5-java-io-exception-when-enabling-the-plugin) for this issue and made some changes that should help, try downloading [this JAR](https://wasabicodes.xyz/cdn/56627d10a69323d87e7c6d3237a38342/xclaim-1.7.0.jar) and see if it is resolved. If not, the output should be at least more helpful this time around.

**Comment by eingruenesbeb** [comment]
Great, I'll make sure to check it out tomorrow 👍🏻

**Comment by WasabiThumb** [comment]
> Great, I'll make sure to check it out tomorrow 👍🏻

Yep, make sure to post a full log with pastebin or similar. Also I just realized, in your original post, it looks like you are using PowerShell? I may be wrong but the colors are spot-on.

**Comment by eingruenesbeb** [comment]
So, I've retested it on Windows (both the version in the main and in the side branch) and I still get the error. However I now think this is rather an issue of an unclean execution environment, rather than a bug on the plugin's behalf. It seems like there is some background process, that prevents access to the file. I also ran a test server on ubuntu, with the otherwise the same setup and voila, I got no exception this time. 
[Here are all the relevant log files](https://pastebin.com/u/Ein_gruenes_Beb/1/j5N98WG2).
So in conclusion:
This is probably only an issue for people like me, who run the server in an unclean environment (so maybe a docker environment would be a solution, but I have no experience with that) for testing purposes or hosting on a local machine, so very niche and there is probably (at least no apparent) way to fix this. I tried looking for the specific process a, that was interfering, for a bit, but no luck yet...

**Comment by WasabiThumb** [comment]
When you say that you tested it "on the side branch", do you mean that you built the JAR from source? The releases tab only contains binaries built on the main branch, the only published build of the side branch was linked above.

**Comment by eingruenesbeb** [comment]
I tested it with the jar, you've linked in this comment section. [This one](https://wasabicodes.xyz/cdn/56627d10a69323d87e7c6d3237a38342/xclaim-1.7.0.jar)

**Comment by WasabiThumb** [comment]
Okay, thanks for clearing that up. That is the one with the changes applied. You should have gotten a more verbose output when testing on Windows that would be useful to see and could potentially fix this problem

**Comment by eingruenesbeb** [comment]
Oh okay, here are the other files... (oops):
https://pastebin.com/LaiznKNh
https://pastebin.com/HnWRPM5j

**Comment by WasabiThumb** [comment]
So according to these logs the problem occurs because XClaim is unable to delete the existing lang file, giving this error (reported by OS presumably):
```
Der Prozess kann nicht auf die Datei zugreifen, da sie von einem anderen Prozess verwendet wird
```
and that does in fact suggest that the file is in use by another process.

There are some tutorials online that detail how to find what process is using a file using Resource Monitor, I would probably look into that. I don't know what would be causing this other than an antivirus scanner though. In either case I believe I can close this issue since it's not strictly plugin related? Unless you find that the process using the file is Java...

**Comment by WasabiThumb** [comment]
There are ways of overwriting a file without deleting it though, that would probably solve this and increase compatibility. I think I will do that

**Comment by eingruenesbeb** [comment]
Yeah, that might be a good idea 👍🏻 
Anyways, yeah... I think this issue can be closed now.
Thanks for the help :D

**Comment by WasabiThumb** [comment]
Should be fixed in 1.7.1
